great_expectations/rule_based_profiler/domain_builder/categorical_column_domain_builder.py

```diff
@@ -390,7 +177,9 @@ def _generate_metric_configurations_to_check_cardinality(
             List of dicts of the form [{column_name: List[MetricConfiguration]},...]
         """
 
-        limit_mode: CardinalityLimitMode = self.cardinality_checker.limit_mode
+        limit_mode: Union[
```

Comment:
Love this, thank you for the catch. Could it also be simplified to just the parent `CardinalityLimit`?